ra2ce/analyses/indirect/analyses_indirect.py

```diff
@@ -995,26 +1013,59 @@ def multi_link_isolated_locations(self, graph, analysis, crs=4326):
                 crs=crs,
             )
 
-            # remove the largest (main) graph
+            # remove the largest (main) graph, leaving isolated network
             results.sort_values(by="count", ascending=False, inplace=True)
             results = results.loc[results["count"] != results["count"].max()]
 
             # reproject the datasets to be able to make a buffer in meters
             results = results.set_crs(crs=crs)
             results.to_crs(crs=nearest_utm, inplace=True)
 
+            # process network to get output
+            # get directly disrupted roads
+            results_flooded_roads = gpd.GeoDataFrame(
+                {
+                    "geometry": [e[-1]["geometry"] for e in edges_remove],
+                    "edge_fid": [
+                        f"{e[-1]['node_A']}_{e[-1]['node_B']}" for e in edges_remove
+                    ],
+                },
+                crs=crs,
+            )
+            results_flooded_roads.to_crs(crs=nearest_utm, inplace=True)
+            results_flooded_roads.geometry = results_flooded_roads.geometry.buffer(
+                analysis["buffer_meters"],
+                cap_style=3,
+            )
+            results_flooded_roads[f"i_type_{hazard_name[:-3]}"] = "flooded"
+
+            # get indirect isolated roads
             results_buffered = results.copy()
             results_buffered.geometry = results_buffered.geometry.buffer(
-                analysis["buffer_meters"]
+                analysis["buffer_meters"],
+                cap_style=3,
             )
+            results_buffered[f"i_type_{hazard_name[:-3]}"] = "isolated"
+
+            # merge for output, set original crs
+            results_buffered = gpd.GeoDataFrame(
+                pd.concat([results_buffered, results_flooded_roads])
+            ).to_crs(crs=crs)
+
+            # Save the output
             results_buffered.to_file(
                 self.config["output"]
                 / analysis["analysis"]
                 / f"isolated_{hazard_name}.shp"
-            )  # Save the buffer
+            )
 
-            intersect = gpd.overlay(results_buffered, locations, keep_geom_type=False)
-            intersect.set_crs(crs=nearest_utm)
+            # relate the locations to network disruption
+            # join the locations on edges_fid and results_buffered on edge_fid
+            joined = locations.merge(
+                results_buffered.drop(columns="geometry"), on="edge_fid", how="left"
+            )
+            intersect = joined.loc[~joined[f"i_type_{hazard_name[:-3]}"].isna()]
```

Comment:
Extract the `get_intersect(...)` functionality (or encapsulate it in a method here), the fact that it's being 'overwritten' in line 1079 it's not a good sign, it's better if you just 'return' that value within a method.

Reply:
a new private function `_get_location_network_intersect` has been added